src/run_environment/machine.ml

```diff
@@ -122,6 +122,11 @@ module Make_fun = struct
                                   ["#WIP"]). *)
       | `Custom of string (** Pass arbitrary data (useful for temporary
                               extensions/experiements outside of Biokepi). *)
+      | `Self_identification of string list
```

Comment:
Another downside: if you have different invocations of the same tool, you can't configure them separately.